packages/components/src/composite/legacy.tsx

```diff
@@ -0,0 +1,174 @@
+/**
+ * Composite is a component that may contain navigable items represented by
+ * CompositeItem. It's inspired by the WAI-ARIA Composite Role and implements
+ * all the keyboard navigation mechanisms to ensure that there's only one
+ * tab stop for the whole Composite element. This means that it can behave as
+ * a roving tabindex or aria-activedescendant container.
+ *
+ * @see https://ariakit.org/components/composite
+ */
+
+/**
+ * WordPress dependencies
+ */
+import {
+	forwardRef,
+	useId as useGeneratedId,
+	useRef,
+	useState,
+} from '@wordpress/element';
+import deprecated from '@wordpress/deprecated';
+
+/**
+ * Internal dependencies
+ */
+import * as Current from './v2';
+
+type Component = ( ...any: any[] ) => any;
```

Comment:
Tried, and it's definitely improved, but I can't seem to escape `any` entirely.

Reply:
I'm ok with it, my impression is that it would take a lot of effort to get rigid type safety in this file, which is probably not worth it in this specific case. It already looks like there is a ton of TS fighting going on which isn't really giving us much actual safety when you look at it from end to end.

I do think it's more idiomatic to use the `React.FunctionComponent< any >` type like Marco suggests though, rather than define a custom object with a `displayName` property.

Reply:
I'm sure at one point I tried `FunctionComponent` and it didn't work, but evidently this is not the case, as this seems to be fine...

```ts
type Component = React.FunctionComponent< any >;
```